vpr/src/pack/cluster_placement.cpp

```diff
@@ -127,6 +127,41 @@ bool get_next_primitive_list(t_cluster_placement_stats* cluster_placement_stats,
     for (i = 0; i < cluster_placement_stats->num_pb_types; i++) {
         if (!cluster_placement_stats->valid_primitives[i].empty()) {
             t_cluster_placement_primitive* cur_cluster_placement_primitive = cluster_placement_stats->valid_primitives[i].begin()->second;
+                
+            auto& atom_ctx = g_vpr_ctx.atom();
```

Comment:
This block of code should have a comment ... what is being done here?  If possible, it would also be good to make this a helper function as this is a long routine.

Reply:
stash the change for this file as its a local modification.